harvest/broker/dummy.py

```diff
@@ -75,9 +111,32 @@ def _generate_fake_stock_data(self):
             volume = max(volume + random.randint(-5, 5), 1)  
             yield open_s, high, low, close, volume
 
+    def fetch_prices(self, 
+        start: dt.datetime, 
+        end: dt.datetime, 
+        interval: str='1MIN', 
+        symbol: str=None):
+
+        oldest_timestamp, _ = self.storage.data_range(symbol, interval)
+        if oldest_timestamp is None or start < oldest_timestamp:
+            if oldest_timestamp is None:
+                oldest_timestamp = end
+            data = self.fetch_price_history(start, oldest_timestamp, interval, symbol)[symbol]
+            self.storage.store(symbol, interval, data)
+
+        _, latest_timestamp = self.storage.data_range(symbol, interval)
+        if latest_timestamp is None or end > latest_timestamp:
+            if latest_timestamp is None:
+                latest_timestamp = start 
+            data = self.fetch_price_history(latest_timestamp, end, interval, symbol)[symbol]
+            self.storage.store(symbol, interval, data)
+
+        return self.storage.load(symbol, interval, start, end)
```

Comment:
I am actually going to remove this since the caching will be done by the trader.

Reply:
Do you think you can also rework the tests and other endpoints in `DummyBroker`? I'm just starting to realize how important this endpoint was to `DummyBroker`